types/module/server/manager.go

```diff
@@ -137,13 +147,59 @@ func (mm *Manager) CompleteInitialization() error {
 	return nil
 }
 
+// InitGenesis performs init genesis functionality for modules.
+// We pass in existing validatorUpdates from the sdk module Manager.InitGenesis.
+func (mm *Manager) InitGenesis(ctx sdk.Context, genesisData map[string]json.RawMessage, validatorUpdates []abci.ValidatorUpdate) abci.ResponseInitChain {
+	for name, initGenesisHandler := range mm.initGenesisHandlers {
+		if genesisData[name] == nil || initGenesisHandler == nil {
+			continue
+		}
+		moduleValUpdates, err := initGenesisHandler(types.Context{Context: ctx}, mm.cdc, genesisData[name])
+		if err != nil {
+			panic(err)
+		}
+
+		// use these validator updates if provided, the module manager assumes
+		// only one module will update the validator set
+		if len(moduleValUpdates) > 0 {
+			if len(validatorUpdates) > 0 {
+				panic("validator InitGenesis updates already set by a previous module")
+			}
+			validatorUpdates = moduleValUpdates
+		}
+	}
+
+	return abci.ResponseInitChain{
+		Validators: validatorUpdates,
+	}
+}
```

Comment:
This server manager is used for the new modules (ecocredit, data, group) which are not registered with the usual sdk manager (see module wiring in `app.go` and `experimental_appconfig.go`). These new modules don't implement `AppModule` and don't have a direct access to the server implementation which is private (contrary to sdk modules with a reference to a keeper). So we pass relevant methods (init/export genesis in this case) to the `configurator` on `RegisterServices` so they can be passed in turn to the server manager on `RegisterModules`, then the server manager can use them to perform init/export genesis when needed.

Reply:
Could you add a doc string explaining that (to the `Manager.Init/Export Genesis`

Reply:
Also, this mechanism looks tricky. Instead of calling interface, we pass a manager and ask the module to register itself. I don't think that extending the Registration pattern is a good approach. Not going to block on that. Let me know wht do you think. If we find that it's good to rethink it then maybe we can continue in a new issue?
@aaronc  - do you have any comments on that?

Reply:
Maybe we should try something like I'm describing here which is a hybrid between the ADR 033 approach and the existing approach: https://github.com/regen-network/regen-ledger/issues/301#issuecomment-804179925.

Reply:
Let's follow up on that as part of this issue https://github.com/regen-network/regen-ledger/issues/301

Reply:
Yes, let's roll this PR and think about better approach in the issue.